src/test/java/org/junitpioneer/jupiter/RetryingTestExtensionTests.java

```diff
@@ -132,6 +134,10 @@ void failsAlways_executedThreeTimes_fails() {
 				.hasNumberOfDynamicallyRegisteredTests(3)
 				.hasNumberOfAbortedTests(2)
 				.hasNumberOfFailedTests(1);
+
+		assertThat(results)
+				.asFailureAssert()
+				.andThenCheckExceptions(throwables -> assertLastExceptionFailures(throwables).hasSize(3));
```

Comment:
```suggestion
		assertThat(results)
				.hasSingleFailedTest()
				.withExceptionInstanceOf(MultipleFailuresError.class)
				.extracting(MultipleFailuresError::getFailures, InstanceOfAssertFactories.LIST)
				.hasSize(3);
```

Reply:
I think this solution is clean enough to avoid expanding `PioneerAssert`. What do you think?

Reply:
Thanks for the hint! Switched to this method and removed my extension of `PioneerAssert`. Also added similar check to all other failed tests.